packages/databricks-sdk-js/src/.codegen/api.ts.tmpl

```diff
@@ -30,88 +31,102 @@ export class {{$s.PascalName}}Service {
 	constructor(readonly client: ApiClient){}
 
 	{{- range $s.Methods}}
-	/**
-    {{.Comment "    * " 80}}
-	*/
+
 	@withLogContext(ExposedLoggers.SDK)
-    async {{.CamelName}}({{if .Request}}request:  model.{{.Request.PascalName}},{{end}}
+    private async _{{.CamelName}}({{if .Request}}request:  model.{{.Request.PascalName}},{{end}}
         @context context?: Context
-	): Promise<{{if .Response}}
-        {{if .Response.ArrayValue}}
-            Array<model.{{.Response.ArrayValue.PascalName}}>
-        {{else}}
-            model.{{.Response.PascalName}}
-        {{end}}
-	{{else}}
-      model.EmptyResponse
-	{{end}}> 
+	): Promise<{{template "type_for_response" .}}> 
 	
 	{
-            const path = {{if .PathParts -}}
-		        `{{range  .PathParts}}{{.Prefix}}{{if .Field}}${request.{{.Field.Name}}}{{ else if .IsAccountId }}${this.client.accountId}{{end}}{{ end }}`
-	        {{- else}}"{{.Path}}"{{end}}
-			return (await this.client.request(
-				path,
-				"{{.Verb}}",
-				{{if .Request -}}request{{else}}undefined{{end}}, 
-				context
-			) as {{if .Response}}{{if .Response.ArrayValue}}Array<model.{{.Response.ArrayValue.PascalName}}>{{else}}model.{{.Response.PascalName}}{{end}}{{else}}model.EmptyResponse{{end}})
-		}
+		const path = {{if .PathParts -}}
```

Comment:
once i push out the SDK generator improvement, you can simplify this code even more. stay tuned.